src/core_modules/capture-core/components/Pages/MainPage/EventWorkingLists/DataSourceSetup/EventWorkingListsDataSourceSetup.component.js

```diff
@@ -5,27 +5,29 @@ import { errorCreator } from 'capture-core-utils';
 import { convertClientToList } from '../../../../../converters';
 import { EventWorkingListsTemplateSetup } from '../TemplateSetup';
 import type { Props } from './eventWorkingListsDataSourceSetup.types';
-import type { EventsMainProperties, EventsDataElementValues } from '../types';
 
 export const EventWorkingListsDataSourceSetup = ({
-    eventsMainProperties,
-    eventsDataElementValues,
+    eventRecords,
     columns,
     recordsOrder,
     ...passOnProps
 }: Props) => {
-    const dataSource = useMemo(() => Object
-        .keys(eventsMainProperties || {})
-        .map(key => ({
-            ...(eventsMainProperties && eventsMainProperties[key]),
-            ...(eventsDataElementValues && eventsDataElementValues[key]),
-        }))
-        // $FlowFixMe
-        .reduce((accDataSource, clientRecord: EventsMainProperties & EventsDataElementValues) => {
+    const eventRecordsArray = useMemo(() =>
+        recordsOrder && eventRecords && recordsOrder
+            .map(eventId => ({
+                ...eventRecords[eventId],
+                eventId,
+            })), [
+        eventRecords,
+        recordsOrder,
+    ]);
```

Comment:
This ties to some comment I made on the previous PR and some comment I make on this one when you use `[]` instead of `null`. 

I think here you can also do `[]` instead of having this value being `undefined`. 

Again I dont know how you see this but in general I think avoiding the following ` someValue?: Array<something>` and instead have  `someValue: Array<?something>`  gives more information.

Reply:
It depends for me. It's also good to know if the data has been fetched or not. For the offline list (upcoming PR) I utilize this information.

Reply:
okay thats a fair point